### PR TITLE
Fix for issue 128

### PR DESCRIPTION
**Pull request by dimsh99**

### CODE REVIEW

File: recipes/server.rb [hunk 1]
```diff
@@ -80,6 +80,14 @@
     action :nothing
   end
 
+  bash 'Restore security context' do
+    user 'root'
+    code "/usr/sbin/restorecon -v #{node['mariadb']['mysqld']['default_datadir']}"
+    only_if '[ -f /usr/sbin/selinuxenabled ] && /usr/sbin/selinuxenabled'
```
Comment (shoekstra) [inline recipes/server.rb]: We should use the `selinux_enabled?` chef helper here instead.

This should be accessible if we add `Chef::Recipe.send(:include, Chef::Util::Selinux)` at the top of the recipe.

**Comment by dimsh99** [comment]
Thanks, Stephen, I'll check on this.

**Comment by dimsh99** [comment]
I remembered why I gave up on using `selinux_enabled?` first time.
If `Chef::Recipe.send(:include, Chef::Util::Selinux)` added to recipe `selinux_enabled?` can't be accessed from within resource only_if guard:
undefined method `selinux_enabled?' for Chef::Resource::Bash

I see 2 ways to make it work:
1. Inject Selinux into bash resource: 
   `Chef::Resource::Bash.send(:include, Chef::Util::Selinux)`
2. Condition around resource:
  `if selinux_enabled?
     bash 'Restore security context' do
     ....
     end
    end`

The first approach makes all Bash resources in runlist "infected" by Chef::Util::Selinux, the other one is considered bad practice in chef.
So I'm not sure what to choose. Maybe I'm missing something?

**Comment by shoekstra** [comment]
Now that you mention that, it does sound familiar. I dug into our wrapper cookbook and we ended up with something like this:

```
extend Chef::Util::Selinux

selinux_on = selinux_enabled?
```

and then use `selinux_on` in the resource's guard block?  I think this would be the cleanest option IMO.

On a separate note we should discuss SELinux testing, but let's do that in #155.

File: recipes/server.rb (hunk 1 above)
Comment (shortdudey123) [reply recipes/server.rb]: Agree on this

**Comment by shoekstra** [comment]
I've rebased and fixed up the guard to test if SELinux is enabled.

Thanks for the contribution!

**Pull request merged**